### ISSUE TITLE
[FEATURE] - Conditional rendering

### ISSUE THREAD

**Issue by cregourd** [premieroctet/next-admin#372]
### Summary

Add possibility to hide/show fields in form based on `(model) => boolean` function in options

### Basic Example

```ts
fields: {
   example: {
      visible: (model) => model.show_example
   }
}
````

**Comment by foyarash** [comment]
Available in v5.3.0

**Comment by 0xCucurbitaceae** [comment]
@foyarash if i read the code correctly, this entirely removes the field from the server response. Would it not be more appropriate to show/hide the field reactively in the frontend? so that if smone checks "Has Metadata", the metadata fields would appear?

In addition to this, i'd argue that the "dependent" fields should be set to null if "Has Metadata" gets unchecked (although it could depend on the usecase)

Thanks!
